### PR TITLE
[feature#14818] alert-spi support FeiShuPersonalWork

### LINKED ISSUE #14818: [Feature] Feishu and Dingding support sending notifications to personal work
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

如果有很多人开发作业，希望每个人开发的作业，告警消息发送个人工作通知中，避免告警消息轰炸

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xjlgod**
## Purpose of the pull request

to add dinding talk personal work notification close https://github.com/apache/dolphinscheduler/issues/14818

Implementation Mechanism：
[enterprise-app](https://open.feishu.cn/document/home/introduction-to-custom-app-development/self-built-application-development-process)
[enable-bot-ability](https://open.dingtalk.com/document/orgapp/chatbot-sends-queries-and-withdraws-one-on-one-chat-messages)
[send-meeage](https://open.feishu.cn/document/server-docs/im-v1/message/create?appId=cli_a59c4fb4e838500e)

## Brief change log

a new moudle dolphinscheduler-alert-feishu-personalwork be added

## Verify this pull request

image

69864f80d46b3ea97f6a7188f6f9410

### CODE REVIEW

**Review by songjianet — DISMISSED** [review]

**Review by SbloodyS — CHANGES_REQUESTED** [review]
Since we already have `dolphinscheduler-alert-feishu` module. I think it's better add this in it.

**Comment by xjlgod** [comment]
> Since we already have `dolphinscheduler-alert-feishu` module. I think it's better add this in it.

Do you mean to put the code in  `dolphinscheduler-alert-feishu` module, or this feishu personal work plugin should also be integrated into feishu plugin. If it were the second one, I think it would make the feishu plugin look bulky and require many parameters to be filled in.

**Comment by SbloodyS** [comment]
Yes. Please putting the code in `dolphinscheduler-alert-feishu` module. @xjlgod

**Comment by xjlgod** [comment]
> Yes. Please putting the code in `dolphinscheduler-alert-feishu` module. @xjlgod

please check it

File: dolphinscheduler-ui/src/locales/en_US/security.ts [hunk 1]
```diff
@@ -267,7 +267,11 @@ export default {
     AtUserIds: 'At User Ids',
     MsgType: 'Msg Type',
     // eslint-disable-next-line quotes
-    IsAtAll: "{'@'}All"
+    IsAtAll: "{'@'}All",
+    appId: 'AppId',
+    appSecret: 'AppSecret',
+    receiveIdType: 'ReceiveIdType',
+    receiveId: 'receiveId'
```
Comment (SbloodyS) [inline dolphinscheduler-ui/src/locales/en_US/security.ts]: Why did you add this?

Comment (xjlgod) [reply dolphinscheduler-ui/src/locales/en_US/security.ts]: This is to display appropriate field names in the UI. For instance, if not add ‘appId’, it will be displayed as ‘alarm_ group.alarm_ Instance.appId’,

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/personalwork/FeiShuPersonalWorkSender.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+/**
+ * <p>
+ *     https://open.feishu.cn/document/home/introduction-to-custom-app-development/self-built-application-development-process
+ *     https://open.feishu.cn/document/faq/trouble-shooting/how-to-enable-bot-ability
+ *     https://open.feishu.cn/document/server-docs/im-v1/message/create?appId=cli_a59c4fb4e838500e
+ * </p>
+ */
+@Slf4j
+public class FeiShuPersonalWorkSender {
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/personalwork/FeiShuPersonalWorkSender.java]: Although this implementation method works, it is recommended to take a look at the implementation of `APP` and `APPCHAT` in `dolphinscheduler-alert-wechat`.

Comment (xjlgod) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/personalwork/FeiShuPersonalWorkSender.java]: I get it, but the parameters for personal notifications are completely different than the existing feishu parameters. Only the type selection parameter (like `APP` and `APPCHAT` ) is changed to required parameters, and the rest is only without required checks. Is this acceptable?

**Comment by xjlgod** [comment]
These two pictures show how to use group notifications and personal work notifications @SbloodyS 
9e69b4d3ccd00de5a0537e6e9c8696f
524f227d3e4a61d41a4c36b4c12bc19
